test/e2e/seeder/anvil-seeder.js

```diff
@@ -17,20 +17,26 @@ class AnvilSeeder {
    * @param contractName
    */
 
-  async deploySmartContract(contractName) {
+  async deploySmartContract(contractName, hardfork) {
     const { publicClient, testClient, walletClient } = this.provider;
     const fromAddress = (await walletClient.getAddresses())[0];
 
     const contractConfig = contractConfiguration[contractName];
     const deployArgs = this.getDeployArgs(contractName, contractConfig);
 
-    const hash = await walletClient.deployContract({
+    const deployOptions = {
       abi: contractConfig.abi,
       account: fromAddress,
       args: deployArgs,
       bytecode: contractConfig.bytecode,
-      gasPrice: 2000000000,
-    });
+    };
+
+    // Add gasPrice if hardfork is muirGlacier to indicate it's a legacy tx
+    if (hardfork === 'muirGlacier') {
+      deployOptions.gasPrice = 20000;
+    }
+
```

Comment:
here we handle the case when we use legacy tx's, if we pass the gasPrice, it reads the tx as legacy

Reply:
I'm curious if `deployOptions.type = "0x0"` would also work?

Reply:
unfortunately it doesn't work :( It would be nice to have a param to indicate the tx type instead of passing the gasPrice, but haven't found it in viem doc. IT seems under the hood the type 2 gas params are always passed :thinking:  The only way I found it to work is to set the gasPrice, then it sets it as legacy tx :/ 

![Screenshot from 2025-03-18 18-36-07](https://github.com/user-attachments/assets/bb6cb603-c39a-4cd9-8c7b-fde665896c33)

Reply:
Apparently you can use `type: 'legacy'`

https://github.com/wevm/viem/discussions/3465#discussioncomment-12544152

Reply:
thank you for your throughtful investigation David :pray: I've attempted to change it to `type: 'legacy'` but it seems the error is the same. Somehow this is not picked up? This is when I'm using the `deployContract` method. Could it be a bug on their side? 

![image](https://github.com/user-attachments/assets/56e3f957-17d9-457b-af36-2e96cc8c3165)

![Screenshot from 2025-03-19 09-44-10](https://github.com/user-attachments/assets/e24a4866-2013-4c34-bb7b-7240f92991b0)

Reply:
Yes, I think it is a bug in viem: https://github.com/wevm/viem/discussions/3465#discussioncomment-12604072

Reply:
thank you v much, David, I've opened a ticket for us to change the gasPrice with the tx type, once the issue is fixed on their side :pray: 

https://github.com/MetaMask/MetaMask-planning/issues/4496